spec/unit/pops/loaders/loaders_spec.rb

```diff
@@ -123,6 +140,68 @@
     expect(loaders.private_environment_loader().to_s).to eql("(DependencyLoader 'environment private' [])")
   end
 
+  it 'creates a hierarchy of loaders' do
+    hierarchy = Puppet::Pops::Loaders.new(empty_test_env).loader_hierarchy
+    expected_loaders = [
+      Puppet::Pops::Loader::StaticLoader,
+      Puppet::Pops::Loader::ModuleLoaders::LibRootedFileBased,
+      Puppet::Pops::Loader::Runtime3TypeLoader,
+      Puppet::Pops::Loader::SimpleEnvironmentLoader,
+      Puppet::Pops::Loader::DependencyLoader
+    ]
+    expected_loader_names = [
+      nil,
+      'puppet_system',
+      empty_test_env.name,
+      'environment',
+      'environment private'
+    ]
+    expect(hierarchy).to contain_exactly(*expected_loaders)
+    expect(hierarchy.map(&:loader_name)).to contain_exactly(*expected_loader_names)
+  end
+
+  it 'excludes the Runtime3TypeLoader when tasks are enabled' do
+    Puppet[:tasks] = true
+    hierarchy = Puppet::Pops::Loaders.new(empty_test_env).loader_hierarchy
+    expected_loaders = [
+      Puppet::Pops::Loader::StaticLoader,
+      Puppet::Pops::Loader::ModuleLoaders::LibRootedFileBased,
+      Puppet::Pops::Loader::ModuleLoaders::EmptyLoader,
```

Comment:
it's odd that when tasks is false, the public environment loader is an instance of `SimpleEnvironmentLoader`, but here it's an `EmptyLoader`, yet both tests are using the same `empty_test_env`...